snap/snapcraft.yaml

```diff
@@ -0,0 +1,49 @@
+name: brogue-ce
+summary: Welcome to the Dungeons of Doom!
+description: Brogue is a single-player strategy game set in the halls of a mysterious and randomly-generated dungeon.
+license: AGPL-3.0
+
+adopt-info: brogue-ce
+confinement: strict
+base: core18
+
+parts:
+  brogue-ce:
+    source: .
+    plugin: make
+
+    override-pull: |
+      snapcraftctl pull
+      version="$(git describe --always | sed -e 's/-/+git/;y/-/./')"
+      [ -n "$(echo $version | grep "+git")" ] && grade=devel || grade=stable
+      snapcraftctl set-version "$version"
+      snapcraftctl set-grade "$grade"
+
+    override-build: |
+      # Build and install brogue - no install target provided
+      make clean && make DATADIR='\$$SNAP/usr/share' &&
+      cp bin/brogue $SNAPCRAFT_PART_INSTALL/usr/bin/ &&
+      cp -r bin/assets $SNAPCRAFT_PART_INSTALL/usr/share/ && chmod -R +rX $SNAPCRAFT_PART_INSTALL/usr/share/assets
+
+      # Configure wrapper to run brogue from user data directory
+      echo 'cd $SNAP_USER_DATA && brogue "$@"' >> $SNAPCRAFT_PART_INSTALL/usr/bin/brogue-wrapper &&
```

Comment:
I think `SNAP_USER_COMMON` would be better, so replays and saves are kept between updates.

Reply:
`SNAP_USER_DATA` is copied between updates - I think the intent is that if a new release does something interesting like, changing the save format, rolling back to an old release will also roll back the saves.